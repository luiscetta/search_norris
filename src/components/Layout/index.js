import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import Homepage from '../../Pages/Homepage';


export default function Layout() {
    return (
        <Router>
            <Homepage />
        </Router>
    );
}