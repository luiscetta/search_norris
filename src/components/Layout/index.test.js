import React from 'react';

import { create } from 'react-test-renderer';

import Layout from './';

function MockedRouter({ children }) {
    return <div>{children}</div>
}

jest.mock('react-router-dom', () => ({
    BrowserRouter: MockedRouter,
    Route: MockedRouter,
    Routes: MockedRouter,
}));


describe('Layout component', () => {
    describe('When the component is rendered', () => {
        it('Should match snapshot', () => {
            const wrapper = create(<Layout />);
            expect(wrapper.toJSON()).toMatchSnapshot();
        })
    })
});