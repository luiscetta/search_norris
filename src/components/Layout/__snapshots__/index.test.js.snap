// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component When the component is rendered Should match snapshot 1`] = `
<div>
  <div
    className="container"
    id="top"
  >
    <div
      className="content"
    >
      <img
        alt="chuck norris with guns"
        src={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      />
      <h1>
        Search Norris
      </h1>
      <div
        className="input-field"
      >
        <label
          htmlFor="input"
        >
          Search for a word (or text)
        </label>
        <input
          autoFocus={true}
          className="search-input"
          onChange={[Function]}
          type="search"
          value=""
        />
      </div>
      <div
        className="btn-content"
      >
        <button
          className="btn-search"
          onClick={[Function]}
        >
          <svg
            className="search-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          Search
        </button>
        <button
          className="btn-lucky"
          onClick={[Function]}
        >
          <svg
            className="lucky-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M105.186 26.426c-38.314.06-62.13 26.334-31.37 72.994C6.192 97.785-9.36 188.75 62.29 218.39c54.838 22.684 82.008-6.012 142.985 15.008-64.015-7.88-119.143 21.696-153.814 66.6-38.507 49.87-24.104 120.148 40.41 100.332-9.024 67.12 62.385 100.137 103.907 47.166 35.793-45.66 15.51-103.756 41.842-182.885 7.1 61.706-5.617 108.762 11.573 166.478 21.393 71.83 103.075 75.605 118.537 6.34 55.84 51.31 95.82-9.41 75.754-76.643-17.993-60.285-90.76-98.2-167.65-118.703 72.808-3.032 108.846 44.51 167.78 30.084 72.446-17.734 64.365-108.426-10.823-128.922 45.767-54.918-19.446-110.047-87.18-93.117-57.113 14.275-90.85 71.977-101.928 143.33-5.55-65.618-28.823-120.138-81.03-150.316-19.64-11.353-40.05-16.74-57.466-16.714z"
            />
          </svg>
          I'm feeling lucky!
        </button>
      </div>
    </div>
  </div>
</div>
`;
